crates/cext/src/transpiler/passes/split_2q_unitaries.rs

```diff
@@ -0,0 +1,184 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use crate::pointers::const_ptr_as_ref;
+
+use qiskit_circuit::circuit_data::CircuitData;
+use qiskit_circuit::converters::dag_to_circuit;
+use qiskit_circuit::dag_circuit::DAGCircuit;
+use qiskit_transpiler::passes::run_split_2q_unitaries;
+
+/// The result from ``qk_transpiler_pass_standalone_split_2q_unitaries()``.
+pub struct Split2qUnitariesResult {
+    circuit: CircuitData,
+    permutation: Vec<u32>,
+}
+
+/// @ingroup QkSplit2qUnitariesResult
+/// Get the modified circuit if any unitaries were split
+///
+/// @param result a pointer to the result.
+///
+/// @returns A pointer to the output circuit.
+///
+/// # Safety
+///
+/// Behavior is undefined if ``result`` is not a valid, non-null pointer to a
+/// ``QkSplit2qUnitariesResult``. The pointer to the returned circuit is owned by
+/// the result object, it should not be passed to ``qk_circuit_free()`` as it
+/// will be freed by ``qk_split_2q_unitaries_result_free()``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_split_2q_unitaries_result_circuit(
+    result: *const Split2qUnitariesResult,
+) -> *const CircuitData {
+    let result = unsafe { const_ptr_as_ref(result) };
+    &result.circuit as *const _
+}
+
+/// @ingroup QkSplit2qUnitariesResult
+/// Get the length of the permutation array of the permutation caused by swap equivalent unitaries
+/// that were split
+///
+/// @param result a pointer to the result.
+///
+/// @returns The length of the permutation array stored in the result. This will be 0 if
+/// ``qk_transpiler_pass_standalone_split_2q_unitaries()`` is called with ``split_swaps`` set to
+/// ``false`` or there were no swap equivalent unitaries in the circuit.
+///
+/// # Safety
+///
+/// Behavior is undefined if ``result`` is not a valid, non-null pointer to a
+/// ``QkSplit2qUnitariesResult``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_split_2q_unitaries_result_permutation_len(
+    result: *const Split2qUnitariesResult,
+) -> usize {
+    let result = unsafe { const_ptr_as_ref(result) };
+    result.permutation.len()
+}
+
+/// @ingroup QkSplit2qUnitariesResult
+/// Get the length of the permutation array of the permutation caused by swap equivalent unitaries
+/// that were split
+///
+/// @param result a pointer to the result.
+///
+/// @returns A pointer to the permutation array with the length that can be found with
+/// ``qk_split_2q_unitaries_result_permutation_len()``.
```

Comment:
Try using sphinx C++ formatting for this and other refs.
```suggestion
/// :cpp:func:`qk_split_2q_unitaries_result_permutation_len()`.
```

Reply:
It should probably be C domain not C++, or we've got Doxygen/Breathe configured wrong.

Reply:
Do we have it setup _at all_ for cross refs? I thought we didn't and it doesn't actually do anything anyways to add the `:c:func:`? 😅

Reply:
This was removed in: https://github.com/Qiskit/qiskit/pull/14720/commits/ae02b37d84f964a49e28b6b23710d4e20b5144ee

Reply:
@Cryoris @jakelishman to answer your question, even though this is resolved, our Sphinx configuration is wrong and we are exposing it as C++, so if you use the C++ markers, it correctly links to each method.